lib/rbeapi/api/users.rb

```diff
@@ -211,13 +211,13 @@ def create(name, nopassword = false, secret = nil, opts = {})
           end
           enc = @encryption_map[enc]
 
-          unless secret
-            fail ArgumentError, 'secret must be specified if nopasswd is false'
+          unless opts[:secret]
+            fail ArgumentError, 'secret must be specified if nopassword is false'
           end
-          cmd << " secret #{enc} #{secret}"
+          cmd << " secret #{enc} #{opts[:secret]}"
         end
         cmds = [cmd]
-        cmds << "username name sshkey #{opts[:sshkey]}" if opts[:sshkey]
+        cmds << "username #{name} sshkey #{opts[:sshkey]}" if opts[:sshkey]
```

Comment:
Nice catch! Unfortunately when you catch a bug like this that slipped through the unit tests you need to figure out why. Looking at spec test spec/unit/rbeapi/api/users it should have failed at line 166 when you ran the spec tests. Did you run all the rbeapi spec tests before generating the pull request?